core/src/commonMain/kotlin/com/ustadmobile/core/controller/ReportEditPresenter.kt

```diff
@@ -0,0 +1,264 @@
+        view.verbFilterList = verbDisplaynOneToManyJoinEditHelper.liveList
+        view.contentFilterList = contentOneToManyJoinEditHelper.liveList
+        view.chartOptions = ChartOptions.values().map { ChartTypeMessageIdOption(it, context) }
+        view.yAxisOptions = YAxisOptions.values().map { YAxisMessageIdOption(it, context) }
+        view.xAxisOptions = XAxisOptions.values().map { XAxisMessageIdOption(it, context) }
+        view.groupOptions = XAxisOptions.values().map { GroupByMessageIdOption(it, context) }
+    }
+
+    override suspend fun onLoadEntityFromDb(db: UmAppDatabase): ReportWithFilters? {
+        val entityUid = arguments[ARG_ENTITY_UID]?.toLong() ?: 0L
+
+        val report = withTimeoutOrNull(2000) {
+            db.reportDao.findByUid(entityUid)
+        } ?: Report()
+
+        val reportFilterList = withTimeout(2000) {
+            db.reportFilterDao.findByReportUid(report.reportUid)
+        }
+
+        val groupMap = reportFilterList.groupBy { it.entityType }
+        val personList = groupMap[ReportFilter.PERSON_FILTER] ?: listOf()
+        val verbList = groupMap[ReportFilter.VERB_FILTER] ?: listOf()
+        val contentUidList = groupMap[ReportFilter.CONTENT_FILTER] ?: listOf()
+
+        personOneToManyJoinEditHelper.liveList.sendValue(personList)
+        verbDisplaynOneToManyJoinEditHelper.liveList.sendValue(verbList)
+        contentOneToManyJoinEditHelper.liveList.sendValue(contentUidList)
+
+        report.fromDate = DateTime.nowLocal().localEndOfDay.utc.unixMillisLong - 7.days.millisecondsLong
+        report.toDate = DateTime.nowLocal().localEndOfDay.utc.unixMillisLong
+
+        return ReportWithFilters(report, reportFilterList)
+    }
+
+    override fun onLoadFromJson(bundle: Map<String, String>): ReportWithFilters? {
+        super.onLoadFromJson(bundle)
+
+        val entityJsonStr = bundle[ARG_ENTITY_JSON]
+        var editEntity: ReportWithFilters
+        if (entityJsonStr != null) {
+            editEntity = Json.parse(ReportWithFilters.serializer(), entityJsonStr)
+        } else {
+            editEntity = ReportWithFilters()
+        }
+
+        return editEntity
+    }
+
+
+    override fun onSaveInstanceState(savedState: MutableMap<String, String>) {
+        super.onSaveInstanceState(savedState)
+        val entityVal = entity
+        savedState.putEntityAsJson(ARG_ENTITY_JSON, null,
+                entityVal)
+    }
+
+    override fun handleClickSave(entity: ReportWithFilters) {
+
+        if (entity.reportTitle.isNullOrEmpty()) {
+            view.titleErrorText = systemImpl.getString(MessageID.field_required_prompt, context)
+            return
+        } else {
+            view.titleErrorText = null
+        }
+
+
+        GlobalScope.launch(doorMainDispatcher()) {
+
+            if (entity.reportUid != 0L) {
+                repo.reportDao.updateAsync(entity)
+
+                repo.reportFilterDao.insertListAsync(personOneToManyJoinEditHelper.entitiesToInsert)
```

Comment:
You could make reportFilterDao implement OneToManyDao. Then you could simply call :
```
personOneToManyJoinEditHelper.commitToDatabase(repo.reportFilterDao) {
   reportFk = entity.reportUid
}
```
As you are calling the same method on a list of things, why not do something like:
```
listOf(personOneToManyJoinEditHelper, verbDisplaynOneToManyJoinEditHelper, contentOneToManyJoinEditHelper).forEach {
   it.commitToDatabase(repo.reportFilterDao)
}

```